parsely/src/main/java/com/parsely/parselyandroid/ParselyTracker.kt

```diff
@@ -0,0 +1,424 @@
+/*
+    Copyright 2016 Parse.ly, Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+package com.parsely.parselyandroid
+
+import android.content.Context
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleEventObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.ProcessLifecycleOwner
+import com.parsely.parselyandroid.Logging.log
+import java.util.UUID
+
+/**
+ * Tracks Parse.ly app views in Android apps
+ *
+ *
+ * Accessed as a singleton. Maintains a queue of pageview events in memory and periodically
+ * flushes the queue to the Parse.ly pixel proxy server.
+ */
+public open class ParselyTracker protected constructor(siteId: String, flushInterval: Int, c: Context) {
```

Comment:
Shouldn't we use an `internal` constructor since the class documentation states that it should be accessed through the singleton?

Reply:
Yes, that's a good point. I've addressed it in another PR https://github.com/Parsely/parsely-android/pull/101/files#diff-1e65066b482c03c14a63d8653fe95e6f2239ced194564de01d5a1b2d97a9e12eR33